src/rotest/management/common/json_parser.py

```diff
@@ -113,34 +121,17 @@ def _encode(self, data):
         if data is None:
             return self._NONE_TYPE
 
-        if isinstance(data, basestring):
-            # Strings are surrounded by "" in order to prevent decoding errors.
-            # Without "", the string '5' will be wrongly decoded as number 5.
-            return data
-
-        if isinstance(data, bool):
-            # objectify identify only lower case (true / false) as bool type
-            return data
-
-        if isinstance(data, Number):
-            return data
-
-        if isinstance(data, dict):
-            return self._encode_dict(data)
-
-        if isinstance(data, (list, tuple)):
-            return self._encode_list(data)
-
-        if isinstance(data, type):
-            return self._encode_class(data)
+        base_types = [basestring, bool, Number]
 
-        if isinstance(data, models.Model):
-            return self._encode_resource_data(data)
+        for encoder_type in base_types:
```

Comment:
no more for.
use instead:
data_type = type(data)
dict[data_type] or dict.get(data_type) etc